satpy/tests/reader_tests/test_olci_nc.py

```diff
@@ -137,6 +137,61 @@ def test_get_mask_with_alternative_items(self, mocked_dataset):
         expected = np.array([True] + [False] * 29).reshape(5, 6)
         np.testing.assert_array_equal(res.values, expected)
 
+
+    @mock.patch("xarray.open_dataset")
+    def test_get_l1b_default_mask(self, mocked_dataset):
+        """Test reading mask datasets from L1B products."""
+        import numpy as np
+        import xarray as xr
+
+        from satpy.readers.olci_nc import NCOLCI1B
+        from satpy.tests.utils import make_dataid
+        mocked_dataset.return_value = xr.Dataset({"quality_flags": (["rows", "columns"],
+                                                           np.array([1 << (x % 32) for x in range(35)]).reshape(5, 7))},
+                                                 coords={"rows": np.arange(5),
+                                                         "columns": np.arange(7)})
+        ds_id = make_dataid(name="mask")
+        filename_info = {"mission_id": "S3A", "dataset_name": "mask", "start_time": 0, "end_time": 0}
+        test = NCOLCI1B("somedir/somefile.nc", filename_info, "c")
```

Comment:
Depending on how bored you are, you could change this to not using mocking but instead create a real file on disk. I realize you're just copying what was here before which is why I won't push it.